lib/formalized.html

```diff
@@ -0,0 +1,126 @@
HDD":4,"Power":5},"className":{"TV":"fa-desktop"},"disabled":true},"wemo":{"typeClass":"wemo","title":"Wemo","subdevices":{"Desk Lamp":"192.168.1.9","Fan":"192.168.1.10"},"groups":{"Office":["Desk Lamp","Fan"]},"className":{"Fan":"fa-asterisk"},"disabled":true},"activeBuilding":{"typeClass":"activeBuilding","title":"Packages","appId":"12345","communityId":"1234","unitNumber":"123","apps":{"Announce":{"id":"announceActiveBuilding","controllerIds":["speech","clientSpeech","clientNotify","gerty"]}},"disabled":true},"rss":{"typeClass":"rss","title":"Gym","host":"crossfitbelltown.com","port":80,"path":"/feed/","maxCount":3,"apps":{"Announce":{"id":"announceRss","controllerIds":["speech","clientSpeech","clientNotify","gerty"]}},"disabled":true},"github":{"typeClass":"github","title":"GitHub","owner":"imbrianj","repo":"switchboard","checkVersion":true,"apps":{"Announce":{"id":"announceGithub","controllerIds":["speech","clientSpeech","clientNotify","gerty"]}},"disabled":true},"twitter":{"typeClass":"twitter","title":"Twitter","consumerKey":"xxxxxxxxxxxxxxxxxxxxxx","accessToken":"xxxxxxxxx-xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx","consumerSecret":"xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx","oauthTokenSecret":"xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx","disabledMarkup":true,"disabled":true},"gerty":{"typeClass":"gerty","title":"Gerty","personality":80,"address":false,"ignoreNegative":true,"corrections":{"bedroom lambs":"bedroom lamps","office which":"office switch","hall life":"hall light","kitchen life":"kitchen light","dining life":"dining light"},"names":{"192.168.1.1":"Brian's Phone","192.168.1.2":"Brian's Computer"},"apps":{"Gerty":{"id":"gerty","macros":{"Watch A Movie":"ps3=PowerOn,sleep,sleep,sleep,sleep,sleep,sleep,sleep,sleep,sleep,sleep,sleep,sleep,sleep,sleep,sleep,sleep,sleep,sleep,sleep,sleep,sleep,sleep,Circle,sleep,sleep,Left;samsung=sleep,sleep,sleep,sleep,sleep,sleep,sleep,sleep,sleep,sleep,sleep,sleep,sleep,sleep,sleep,sleep,HDMI4,sleep,DOWN,sleep,ENTER,sleep,sleep,sleep,sleep,ENTER","Goodnight":"smartthings=subdevice-mode-Night","Good Night":"smartthings=subdevice-mode-Night","Goodbye":"smartthings=subdevice-mode-Away;nest=Away","Welcome Home":"smartthings=subdevice-mode-Home;nest=Home"},"controllerIds":["samsung","roku","ps3","panasonic","lg","pioneer","denon","speech","stocks","weather","foscam","mp3","sms","pushover","smartthings","nest","switchBoardCI","xbmc","raspberryRemote","wemo","activeBuilding","clientMp3","clientNotify","clientSpeech"]}},"disabled":true},"debug":{"typeClass":"debug","title":"Debug","disabled":true,"disabledMarkup":true},"formalize":{"whitelist":["welcome","clientMp3","clientNotify","clientSpeech","samsung","roku",{"ps3":["deviceMac","serviceIp","servicePort"]},"panasonic","lg","pioneer","denon",{"speech":"voice"},{"foscam":["username","password"]},"mp3"],"commonFields":["title","typeClass","deviceNotes","disabled","disabledMarkup","deviceIp"]}};</script><script src="updateConfig.js"></script></body></html>
```

Comment:
One big concern is that we do not want to pollute the SB.spec.state object with config values - as that will expose credentials to all connected clients.  So, we could potentially store sensitive info (usernames, passwords, tokens, keys) - and just null them out in the state.  When those values are changed on the client, have them update the config - then null out again before writing to state.